builtin/logical/aws/secret_access_keys.go

```diff
@@ -43,26 +47,41 @@ func secretAccessKeys(b *backend) *framework.Secret {
 	}
 }
 
-func genUsername(displayName, policyName, userType string) (ret string, warning string) {
-	var midString string
+func genUsername(displayName, policyName, userType, usernameTemplate string) (ret string, warning string) {
 
+	var username string
 	switch userType {
 	case "iam_user":
 		// IAM users are capped at 64 chars; this leaves, after the beginning and
 		// end added below, 42 chars to play with.
-		midString = fmt.Sprintf("%s-%s-",
-			normalizeDisplayName(displayName),
-			normalizeDisplayName(policyName))
-		if len(midString) > 42 {
-			midString = midString[0:42]
+		up, err := template.NewTemplate(template.Template(usernameTemplate))
+		if err != nil {
+			return "", fmt.Sprintf("unable to initialize username template: %s", err)
+		}
+
+		um := UsernameMetadata{
+			DisplayName: normalizeDisplayName(displayName),
+			PolicyName:  normalizeDisplayName(policyName),
+		}
+
+		username, err = up.Generate(um)
+		if err != nil {
+			return "", fmt.Sprintf("failed to generate username: %s", err)
+		}
+		fmt.Printf("username: %s\n", username)
```

Comment:
This is still a draft so maybe this is still here intentionally?

Reply:
Forgot about that one! There's still some polishing required in this file, but I'll go ahead and remove that since that particular function works as expected